spec/controllers/users_controller_spec.rb

```diff
@@ -2,11 +2,43 @@
 
 RSpec.describe UsersController, type: :controller do
 
+  before(:each) do
+    request.env["HTTP_REFERER"] = root_url
+    @user = User.new(:email => 'el@el.com', :password => '1234567')
+    @user.save
+    # @user = User.create!(email: "test@test.com", password: 'testtest')
+  end
+
   describe "GET #index" do
     it "returns http success" do
       get :index
       expect(response).to have_http_status(:success)
     end
   end
 
+  describe "GET #show" do
+    it "assign user return as @user" do
+      get :show, params: { id: @user.id }
+    end
+  end
+
+  describe "GET #edit" do
+    it "assigns the requested user as @user" do
+      get :edit, params: { id: @user.id }
+      assigns(:user).should eq(@user)
+    end
+  end
+
+  describe "PUT update" do
+    it "updates the requested user" do
+      put :update, :params => {:id => @user.id, :user => @user.attributes = { :email => "new@new.com", :password => "1234567" }}
+      expect(@user.email).to eq("new@new.com") 
+      expect(response).to redirect_to root_url
+      # expect(response).to redirect_to ('/')
+    end
```

Comment:
how about something like this?

```
let(:update_params) { <write the correct params in here> }

it "updates the requested user" do
  expect { put :update, params: update_params }.to change { @user.email }.from("el@el.com").to("new@new.com")
end
```
Moving the params to a let block allows you to use this one liner without the line becoming too long.  The one liner ensures the correct transition is made; from the old state to the new state.